packages/common/src/project/load.ts

```diff
@@ -3,42 +3,55 @@
 
 import fs from 'fs';
 import path from 'path';
+import {NodeVM, VMScript} from '@subql/x-vm2';
 import {plainToClass} from 'class-transformer';
 import {validateSync} from 'class-validator';
 import yaml from 'js-yaml';
 import {ChainTypes} from './models';
 import {ProjectManifestVersioned, VersionedProjectManifest} from './versioned';
 
-export function loadFromJsonOrYaml(file: string): unknown {
-  const {ext} = path.parse(file);
+export function loadFromFile(filePath: string): unknown {
```

Comment:
Doesn't feel right here. Common is shared between subql-node and subql-query, but apparently subql-query doesn't need to use sandbox to load the js chain type.

Reply:
What `loadFromFile()` here concerns is read a file and return its content. It is subql-node to decide how to use the file content, in this case if it's a .js, subql-node need to interpret it via sandbox.
So you can see  the dependency of `@subql/x-vm2` is always in subql-node, shouldn't be extended to common.

Reply:
The reason I did this is validator/src/readers/local-reader.ts used `loadFromJsonOrYaml()`

Reply:
The reason I didn't do it this way is that validator/src/readers/local-reader.ts also uses `loadFromFile()`. It may not be necessary for validator to be able to do this. Let me know.

Reply:
The validator is not an essential component, if that's the case we can skip validating chaintypes for js like chaintype.

Reply:
let's leave it in common for now. But you need to add more restrict rules for the sandbox.

Reply:
sure, just wanted to get some feedback.